lib/assessment.js

```diff
@@ -1,47 +1,69 @@
 'use strict';
 
-// All of the callbacks in questions 3 through 5 should
-// be called with the current element in the array,
-// its index, and the array itself
-// e.g. callback(currentValue, index, array)
-
 // Question 1
-// Write a callback for reduce that selects
-// the minimum value in the array
+// Write a function that takes zero or more arguments
+// and returns the minimum from that set
+// assume that all the argument are numbers
 var min = function min() {
-
+  var minimum = arguments[0];
+  for (var i = 0; i < arguments.length; i++) {
+    if (arguments[i] < minimum) {
+      minimum = arguments[i];
+    }
+  }
+  return minimum;
 };
 
 // Question 2
-// Write a callback for reduce that produces
-// the sum of the values in the array
+// Write a function that takes zero or more arguments
+// and returns the sum of that set
+// assume that all the argument are numbers
 var sum = function sum() {
-
+  var total = 0;
+  for (var i = 0; i < arguments.length; i++) {
+    total += arguments[i];
+  }
+  return total;
 };
 
 // Question 3
-// Write a function that returns true
-// if invoking the callback for any element
-// in the array returns a truthy value
-var any = function any(array, callback, thisArg) {
-
+// Write a function that returns an array of the specified length
+// with all the elements set to defaultValue
+var newArray = function newArray(length, defaultValue) {
+  ary = []
+  for (var i = 0; i < length; i++) {
+    ary[i] = defaultValue;
+  }
```

Comment:
This function doesn't return anything. If it had returned `ary`, then it would have worked imperfectly; if it had both returned `ary` and declared it before assigning to it, it would be been an ideal solution.

1/2